redbot/core/utils/chat_formatting.py

```diff
@@ -429,3 +430,29 @@ def humanize_timedelta(
             strings.append(f"{period_value} {unit}")
 
     return ", ".join(strings)
+
+
+def text_to_file(
+    text: str, filename: str = "file.txt", *, spoiler: bool = False, encoding: str = "utf-8"
+):
+    """Prepares text to be sent as a file on Discord, without character limit.
+
+    This writes text into a bytes object that can be used for the ``file`` or ``files`` parameters\
```

Comment:
Do not escape line-endings, anywhere in Python, except when using `textwrap.dedent`

Simply limit docstring line length to about 80 characters, use normal newlines, and sphinx / whichever IDE you're using will render the newlines as spaces where appropriate.